src/daft-micropartition/src/micropartition.rs

```diff
@@ -504,6 +506,40 @@ impl MicroPartition {
             unreachable!()
         }
     }
+
+    pub(crate) fn add_monotonically_increasing_id(
+        &self,
+        partition_num: u64,
+        column_name: &str,
+    ) -> DaftResult<Self> {
+        let io_stats = IOStatsContext::new("MicroPartition::add_monotonically_increasing_id");
+        let tables = self.tables_or_read(io_stats)?;
+
+        let tables_with_id = tables
+            .iter()
+            .scan(0u64, |offset, table| {
+                let table_with_id =
+                    table.add_monotonically_increasing_id(partition_num, *offset, column_name);
```

Comment:
I think this is fine, but I wonder if this needs to be a `Table` method, or the `MicroPartition` can be responsible for creating a Series and just concatting it to the Table.

This is I think maybe the first function in which the API differs between the MicroPartition and Table, since we attempted our Table->MicroPartition migration 😛

Reply:
i think its ok too. since the id series creation and concat logic is on a per table basis, having this logic on the micropartition side will introduce some code duplication